src/tensors/braidingtensor.jl

```diff
@@ -126,39 +126,48 @@ end
 TensorMap(b::BraidingTensor) = copy!(similar(b), b)
 Base.convert(::Type{TensorMap}, b::BraidingTensor) = TensorMap(b)
 
+# TODO: fix this!
+# block(b::BraidingTensor, s::Sector) = block(TensorMap(b), s)
 function block(b::BraidingTensor, s::Sector)
     sectortype(b) == typeof(s) || throw(SectorMismatch())
-    (V1, V2) = domain(b)
-    if sectortype(b) == Trivial
+
+    # TODO: probably always square?
+    m = blockdim(codomain(b), s)
+    n = blockdim(domain(b), s)
+    data = Matrix{eltype(b)}(undef, (m, n))
+
+    length(data) == 0 && return data # s ∉ blocksectors(b)
+
+    data = fill!(data, zero(scalartype(b)))
```

Comment:
It is a bit odd looking that here you use `scalartype`, whereas to create `data` you used `eltype`.